pkg/engine/mutation.go

```diff
@@ -101,25 +102,44 @@ func Mutate(policyContext PolicyContext) (resp response.EngineResponse) {
 		if reflect.DeepEqual(policyContext.AdmissionInfo, kyverno.RequestInfo{}) {
 			continue
 		}
+	}
 
-		if strings.Contains(PodControllers, resource.GetKind()) {
+	if strings.Contains(PodControllers, resource.GetKind()) {
+		if !patchedResourceHasPodControllerAnnotation(patchedResource) {
 			var ruleResponse response.RuleResponse
-			ruleResponse, patchedResource = mutate.ProcessOverlay(logger, rule.Name, podTemplateRule, patchedResource)
+			ruleResponse, patchedResource = mutate.ProcessOverlay(logger, "podControllerAnnotation", podTemplateRule.Mutation.Overlay, patchedResource)
 			if !ruleResponse.Success {
 				logger.Info("failed to insert annotation for podTemplate", "error", ruleResponse.Message)
-				continue
-			}
-
-			if ruleResponse.Success && ruleResponse.Patches != nil {
-				logger.V(2).Info("inserted annotation for podTemplate")
-				resp.PolicyResponse.Rules = append(resp.PolicyResponse.Rules, ruleResponse)
+			} else {
+				if ruleResponse.Success && ruleResponse.Patches != nil {
+					logger.V(2).Info("inserted annotation for podTemplate")
+					resp.PolicyResponse.Rules = append(resp.PolicyResponse.Rules, ruleResponse)
+				}
 			}
 		}
 	}
 	// send the patched resource
 	resp.PatchedResource = patchedResource
 	return resp
 }
+
+func patchedResourceHasPodControllerAnnotation(resource unstructured.Unstructured) bool {
+	var podController struct {
+		Spec struct {
+			Template struct {
+				Metadata struct {
+					Annotations map[string]interface{} `json:"annotations"`
+				} `json:"metadata"`
+			} `json:"template"`
+		} `json:"spec"`
+	}
+
+	resourceRaw, _ := json.Marshal(resource.Object)
+	_ = json.Unmarshal(resourceRaw, &podController)
+
+	_, ok := podController.Spec.Template.Metadata.Annotations[PodTemplateAnnotation]
```

Comment:
You can use `patchedResource.GetAnnotations()` to get the annotation map, then verify if `PodTemplateAnnotation` exist.

Reply:
GetAnnotations() will get the annotation of the pod controller, however we want to get the annotations of the pods the pod controller will create